packages/react-divider/src/components/Divider/useDivider.ts

```diff
@@ -1,36 +1,42 @@
 import * as React from 'react';
-import { makeMergeProps, resolveShorthandProps, useId, useMergedRefs } from '@fluentui/react-utilities';
-import { DividerProps, DividerState } from './Divider.types';
+import { getNativeElementProps, resolveShorthand, useId } from '@fluentui/react-utilities';
+import { DividerProps, DividerSlots, DividerState } from './Divider.types';
 
 /**
- * Consts listing which props are shorthand props.
+ * Const listing which props are shorthand props.
  */
-export const dividerShorthandProps = ['wrapper', 'children'] as const;
-
-const mergeProps = makeMergeProps<DividerState>({ deepMerge: dividerShorthandProps });
+export const dividerShorthandProps: Array<keyof DividerSlots> = ['root', 'wrapper'];
 
 /**
  * Returns the props and state required to render the component
  * @param props - Divider properties
  * @param ref - reference to root HTMLElement of Divider
- * @param defaultProps - default values for the properties of Divider
  */
-export const useDivider = (
-  props: DividerProps,
-  ref: React.Ref<HTMLElement>,
-  defaultProps?: DividerProps,
-): DividerState => {
+export const useDivider = (props: DividerProps, ref: React.Ref<HTMLElement>): DividerState => {
+  const { alignContent = 'center', inset = false, vertical = false, appearance, wrapper } = props;
   const dividerId = useId('divider-');
-  const state = mergeProps(
-    {
-      ref: useMergedRefs(ref, React.useRef(null)),
-      /* The Id created to expose accessibility for readers */
-      'aria-labelledby': props.children ? dividerId : undefined,
-      wrapper: { as: 'div', children: props.children, id: dividerId },
-    },
-    defaultProps && resolveShorthandProps(defaultProps, dividerShorthandProps),
-    resolveShorthandProps(props, dividerShorthandProps),
-  );
 
-  return state;
+  return {
+    alignContent,
+    components: {
+      root: 'div',
+      wrapper: 'div',
+    },
+    inset,
+    root: getNativeElementProps('div', {
+      ref,
+      role: 'separator',
+      'aria-labelledby': props.children ? dividerId : undefined,
+      ...props,
+    }),
+    vertical,
+    appearance,
+    wrapper: resolveShorthand(wrapper, {
+      required: true,
+      defaultProps: {
+        id: dividerId,
+        children: props.children,
+      },
+    }),
+  };
```

Comment:
nit: We can order these so that props are passed in first, then we define components and finally we resolve slots.
```suggestion
  return {
    alignContent,
    appearance,
    inset,
    vertical,
    
    components: {
      root: 'div',
      wrapper: 'div',
    },
 
    root: getNativeElementProps('div', {
      ref,
      role: 'separator',
      'aria-labelledby': props.children ? dividerId : undefined,
      ...props,
    }),
    wrapper: resolveShorthand(wrapper, {
      required: true,
      defaultProps: {
        id: dividerId,
        children: props.children,
      },
    }),
  };
```